ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,1094 @@
+                "\n        (var filterMapping, var orderedDropDownValues) = await workspace.GetAnnualCohortAutocompleteMappings();",
+                "\n        FilterMapping[nameof(ReportVariable.AnnualCohort)] = filterMapping;",
+                "\n        return orderedDropDownValues;",
+                "\n    }",
+                "\n",
+                "\n    async Task<IReadOnlyCollection<string>> GetBasicFilterAsync(string userInput, int page, int pageSize) =>",
+                "\n        new string[]{ null }.Concat(",
+                "\n            (FilterName switch",
+                "\n            {",
+                "\n                //GetAutocompleteMappings",
+                "\n               nameof(ReportVariable.Portfolio) => await GetFilterAutocompleteAsync<Portfolio>(),",
+                "\n               nameof(ReportVariable.GroupOfContract) => await GetFilterAutocompleteAsync<GroupOfContract>(),",
+                "\n               nameof(ReportVariable.LineOfBusiness) => await GetFilterAutocompleteAsync<LineOfBusiness>(),",
+                "\n               //nameof(ReportVariable.AnnualCohort) => await GetAnnualCohortFilterAutocompleteAsync(),//TODO the filter is not applied because the prop is an Int",
+                "\n               nameof(ReportVariable.LiabilityType) => await GetFilterAutocompleteAsync<LiabilityType>(),",
+                "\n               nameof(ReportVariable.ValuationApproach) => await GetFilterAutocompleteAsync<ValuationApproach>(),",
+                "\n               nameof(ReportVariable.OciType) => await GetFilterAutocompleteAsync<OciType>(),",
+                "\n               nameof(ReportVariable.InitialProfitability) => await GetFilterAutocompleteAsync<Profitability>(),",
+                "\n",
+                "\n               nameof(ReportVariable.Novelty) => await GetFilterAutocompleteAsync<Novelty>(),",
+                "\n               nameof(ReportVariable.VariableType) => await GetFilterAutocompleteAsync<VariableType>(),",
+                "\n               nameof(ReportVariable.EconomicBasis) => await GetFilterAutocompleteAsync<EconomicBasis>(),",
+                "\n               nameof(ReportVariable.AmountType) => await GetFilterAutocompleteAsync<AmountType>(),",
+                "\n               nameof(ReportVariable.EstimateType) => await GetFilterAutocompleteAsync<EstimateType>(),",
+                "\n                 _ => Enumerable.Empty<string>().ToArray()",
+                "\n            }).Where(x => userInput == null || x.Contains(userInput, StringComparison.OrdinalIgnoreCase)).OrderBy(x => x)).ToArray();",
+                "\n        ",
+                "\n    [NotVisible] IReadOnlyCollection<(string filterName, string filterValue)> InputDataFilter { get; set; }",
+                "\n    ",
+                "\n    //The cast is needed by the .Filter func",
```

Comment:
You mean cast to object? I would remove this comment, or at least add it next to the code.